.github/workflows/build.yml

```diff
@@ -326,11 +337,38 @@ jobs:
     name: Trigger Deployment
     needs: [build, lint, test]
     runs-on: ubuntu-latest
-    if: ${{ success() && github.ref == 'refs/heads/main' }}
     steps:
       - name: Trigger Deployment to QA and Staging
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
         uses: benc-uk/workflow-dispatch@v1.1
         with:
           workflow: Deploy
           token: ${{ secrets.ACTIONS_API_ACCESS_TOKEN }}
           inputs: '{"qa": "true", "staging": "true", "research": "true", "sha": "${{ needs.build.outputs.IMAGE_TAG }}"}'
+
+      - name: Wait for review app deployment
+        id: wait_for_review_app_deployment
+        if: github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'deploy')
+        uses: fountainhead/action-wait-for-check@v1.0.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ github.head_ref }}
+          checkName: ${{ github.event.pull_request.number }} Deployment
+          timeoutSeconds:  1800
+          intervalSeconds: 10
+
+      - name: Exit if review app deployment failed
+        if: ${{ steps.wait_for_review_app_deployment.outputs.conclusion != '' && steps.wait_for_review_app_deployment.outputs.conclusion != 'success' }}
+        run: exit 1
+
+      - name: 'Notify #twd_apply_tech on failure'
```

Comment:
Not sure about this notification, we already notify slack on build and deploy failures on main elsewhere?

Reply:
I'll have a chat with @saliceti about this one after lunch 👍

Reply:
Yes this job doesn't know about deployment failures as it only triggers the `Deploy` workflow but doesn't wait for it. This notification would not be triggered so I guess we don't need it.

Reply:
Winner. I will remove it now.